packages/okta-vue/src/components/ImplicitCallback.vue

```diff
@@ -1,12 +1,31 @@
 <script>
 export default {
   name: 'ImplicitCallback',
-  async beforeMount () {
-    await this.$auth.handleAuthentication()
-    this.$router.replace({
-      path: this.$auth.getFromUri()
-    })
+  data: () => ({
+    error_message: '',
+    error: false,
+  }),
+  async beforeMount() {
+    try {
+      await this.$auth.handleAuthentication();
+      this.$router.replace({
+        path: this.$auth.getFromUri(),
+      });
+    } catch (e) {
+      const handler = this.$auth.onAuthenticationError;
+      if (handler) {
+        return handler.call(this, e);
+      }
+      // Display error message on this callback view
+      this.error_message = e.message;
+      this.error = true;
+    }
   },
-  render () {}
-}
+  render(createElement) {
```

Comment:
I'm unsure about the default handler case here, and genuinely feel we should avoid creating elements on the fly. While this is a fallback mechanism for devs who do not implement their own handler, it still puts users in a bad state (non-branded UI and difficult to read error message).

As an alternative, it might make more sense to emit a [custom event](https://vuejs.org/v2/guide/components-custom-events.html) with some redirect logic. In the default case, we could:

1. Listen to the `OAuthRedirectError` event
2. If the event is triggered, navigate to the applications root path `/`  and log the message via console output

Reply:
IMO it's better to surface the error to the end user if it happens.  A common scenario is this: someone changes the services configuration (e.g. removing a group from the application) and thus accidentally breaks the application for end-users.  If the end user can't see the error, all they can simply report to their support channel is "I'm trying to login and it's not working".  But if they can see the error and report that to their support channel, the resolution can be found more quickly.  

Bump @nbarbettini on this one for your thoughts.  This decision can be applied elsewhere.

Reply:
I'm not proposing we hide errors from the end user, we just need to specify where we want these errors to output to.

Thinking about this a bit more, IMO these libraries shouldn't dictate how an error message gets to the end user. The responsibility of these libs is to catch any/all errors and expose them in a consumable way.

If we take an event approach, we're able to connect multiple error cases to a single handler, as opposed to telling developers to handle each case independently.